components/textbox/textbox_links.jsx

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Link} from 'react-router-dom';
+
+import Constants from 'utils/constants.jsx';
+import * as Utils from 'utils/utils.jsx';
+
+const PreReleaseFeatures = Constants.PRE_RELEASE_FEATURES;
+
+export default class TextboxLinks extends React.Component {
+    static propTypes = {
+        preview: PropTypes.bool,
+        characterLimit: PropTypes.number.isRequired,
+        previewMessageLink: PropTypes.string,
+        updateState: PropTypes.func,
+        message: PropTypes.string.isRequired,
+    };
+
+    static defaultProps = {
+        message: '',
+    };
+
+    togglePreview = (e) => {
+        e.preventDefault();
+        this.props.updateState(!this.props.preview);
+    }
+
+    render() {
+        const hasText = this.props.message && this.props.message.length > 0;
+        let editHeader;
+
+        let helpTextClass = '';
+
+        if (this.props.message && this.props.message.length > this.props.characterLimit) {
+            helpTextClass = 'hidden';
+        }
+
+        if (this.props.previewMessageLink) {
+            editHeader = (
+                <span>
+                    {this.props.previewMessageLink}
+                </span>
+            );
+        } else {
+            editHeader = (
+                <FormattedMessage
+                    id='textbox.edit'
+                    defaultMessage='Edit message'
+                />
+            );
+        }
+
+        let previewLink = null;
+        if (Utils.isFeatureEnabled(PreReleaseFeatures.MARKDOWN_PREVIEW)) {
+            previewLink = (
+                <button
+                    id='previewLink'
+                    onClick={this.togglePreview}
+                    className='style--none textbox-preview-link color--link'
+                >
+                    {this.props.preview ? (
+                        editHeader
+                    ) : (
+                        <FormattedMessage
+                            id='textbox.preview'
+                            defaultMessage='Preview'
+                        />
+                    )}
```

Comment:
This conditional is confusing.  It's like: if preview, show edit header; if not preview, then show preview.

Reply:
nvm and thanks for checking.